### PR TITLE
Feature/styled tooltips

### LINKED ISSUE #134: Add stylized tooltips (what is now the title attributes)

### PR DESCRIPTION
**Pull request by SeanLeRoy**
# Description
This replaces the default tooltips that come with the `title` attribute that the browser will automatically render for us with a styled tooltip component.

Also included, I removed some random old TODOs as well & added some debug information that was annoying me.

## Related Issue
Resolves #134

### CODE REVIEW

File: packages/core/components/Modal/BaseModal/index.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default function BaseModal(props: BaseModalProps) {
                         {title}
                     </h3>
                 ) : null}
-                <TertiaryButton iconName="Cancel" onClick={onDismiss} title="Close" />
+                <TertiaryButton iconName="Cancel" onClick={onDismiss} title="" />
```
Comment (kmitcham) [inline packages/core/components/Modal/BaseModal/index.tsx:45]: Is the empty title here intentional?

**Review by kmitcham — APPROVED** [review]

File: packages/core/components/ListPicker/index.tsx [hunk 2]
```diff
@@ -120,38 +121,42 @@ export default function ListPicker(props: ListPickerProps) {
...
+                                className={classNames(
+                                    {
+                                        [styles.disabled]: !hasUnselectedItem,
+                                    },
+                                    styles.selectAllButton,
+                                    styles.actionButton
+                                )}
+                                disabled={!hasUnselectedItem}
+                                iconProps={{ iconName: "MultiSelect" }}
+                                title={hasUnselectedItem ? undefined : "All options selected"}
```
Comment (aswallace) [inline packages/core/components/ListPicker/index.tsx:136]: Spotting several instances of title still but in other replaces removed; is the tooltip component intended to be a direct replacement?

File: packages/core/components/FileRow/Cell.tsx [hunk 3]
```diff
@@ -94,37 +96,39 @@ export default class Cell extends React.Component<React.PropsWithChildren<CellPr
...
+                        minWidth: Cell.MINIMUM_WIDTH,
+                    }}
+                >
+                    {this.props.children}
+                    <span
+                        className={classNames(styles.resizeTarget, resizeTargetClassName)}
+                        ref={this.resizeTarget}
+                    />
+                </div>
+            </Tooltip>
```
Comment (aswallace) [inline packages/core/components/FileRow/Cell.tsx:116]: Screenshot 2024-09-18 at 10 54 33 AM
Possibly affecting styling here

File: packages/core/components/ListPicker/index.tsx (hunk 2 above)
Comment (SeanLeRoy) [reply packages/core/components/ListPicker/index.tsx]: Ah I seem to have missed a couple, should be caught in https://github.com/AllenInstitute/biofile-finder/pull/241/commits/056a3a915788837a9750eaa5a80191919d29f503, but lmk if you are seeing more

File: packages/core/components/FileRow/Cell.tsx (hunk 3 above)
Comment (SeanLeRoy) [reply packages/core/components/FileRow/Cell.tsx]: Thought this had gone away. Fixed here https://github.com/AllenInstitute/biofile-finder/pull/241/commits/ddc96fa7065f40e713d87c5c000f779acb642104

File: packages/core/components/ListPicker/index.tsx (hunk 2 above)
Comment (aswallace) [reply packages/core/components/ListPicker/index.tsx]: I think the only other one I'm seeing is in ListRow.tsx

**Review by aswallace — APPROVED** [review]

**Review by BrianWhitneyAI — APPROVED** [review]

File: packages/core/components/Modal/BaseModal/index.tsx (hunk 1 above)
Comment (SeanLeRoy) [reply packages/core/components/Modal/BaseModal/index.tsx:45]: Yep

**Pull request merged**